Add support for tree-hierarchized settings groups

Athozus opened the pull request:
Hi,

This is not a big change, but it might be used in the future.

There are two commits, I prefer non-squash for this one : 
* Add a support for parent settings, in `ui/settings.lua` it generates it with a recursive function (not so bad because it is not intended to get thousands of categories)
* Move existing `Default sorting field/direction` to new group `Fields`, child of `Message list`

NB : I'm gonna do a new fields-related setting, so of course this was in my "roadmap" (not famous for being linear lol)
NB2 : I plan to move settings and other non-mail related stuff to new general mods.

![screenshot_20240407_165640](https://github.com/mt-mods/mail/assets/75171564/c6a7c098-d632-4cd7-9110-5d8ff8baffef)

Athozus commented:
@S-S-X I did what you suggested me in our channels in a43f5c3.

IMHO rebase is better for this one, I just have to trick and move commit 430f00a first (same order for others commits), as pre-requisite.

S-S-X commented:
> IMHO rebase is better for this one

Isn't this basically doing one simple thing so why do you think that 4 commits are needed in history?
If it is about luacheck config wouldn't it even then be better to first squash into 2 commits, one for luacheckrc and other for code?

Athozus commented:
> > IMHO rebase is better for this one
> 
> Isn't this basically doing one simple thing so why do you think that 4 commits are needed in history? If it is about luacheck config wouldn't it even then be better to first squash into 2 commits, one for luacheckrc and other for code?

One squash technically possible, for 7f981a4 and a43f5c3, because it concerns the same thing. But I like to dissociate the support addition and later the usage of it.

Athozus commented:
Force push details :
* Moved pre-requisite commit `luacheck` and `table.insert_all()` in first place ;
* Squashed `Add support for tree-hierarchized settings groups` 7f981a4 (11d728b after rebase) and `Compute once the ordered settings groups list` a43f5c3 (2b3e057 after rebase).
* Rebase on `master`
